base/src/dispatcher/dispatcher.ts

```diff
@@ -482,9 +612,59 @@ export class Dispatcher implements ZLUX.Dispatcher {
       return outputObject;
    }        
 
-   makeAction(id: string, defaultName: string, targetMode: ActionTargetMode, type: ActionType, targetPluginID: string, primaryArgument: any):Action{
-     return new Action(id,defaultName,targetMode,type,targetPluginID,primaryArgument);
-   }
+  makeActionFromObject(actionObject: ZLUX.AbstractAction): AbstractAction {
+    const { id, objectType, defaultName } = actionObject;
+    switch (objectType as any) {
+      case "ActionContainer":
+        const container = new ActionContainer(id, defaultName);
+        const actionObjectAsContainer: ZLUX.ActionContainer = actionObject as ZLUX.ActionContainer;
+        const childrenIn: (ZLUX.AbstractAction | ZLUX.ActionReference)[] = actionObjectAsContainer.children;
+        const children: (ZLUX.AbstractAction | ZLUX.ActionReference)[] = [];
+
+        if (childrenIn && childrenIn.length > 0) {
+          for (let childIn of childrenIn) {
+            let child: ZLUX.AbstractAction | ZLUX.ActionReference;
+            if (childIn.hasOwnProperty('targetActionId')) {
+              child = childIn as ZLUX.ActionReference;
+            } else {
+              child = this.makeActionFromObject(childIn as ZLUX.AbstractAction);
+              this.registerAction(child as AbstractAction);
+            }
+            children.push(child);
+          }
+          container.children = children;
+        }
+
+        return container;
+
+      default:
+        const actionAsAny: any = actionObject as any;
+        let { targetMode, type, targetPluginID, primaryArgument } = actionAsAny;
+        // backward compatbility: some "legacy" actions could have a "mode" field instead of a "targetMode" field, or a targetId field instead of a targetPluginID
+        if (targetMode === undefined || targetMode === null) {
+          targetMode = actionAsAny.mode;
+        }
+        if (!targetPluginID) {
+          targetPluginID = actionAsAny.targetId;
+        }
+        return new Action(id, defaultName, targetMode, type, targetPluginID, primaryArgument);
+    }
+
+  }
+
+  makeAction(id: string, defaultName: string, targetMode: ActionTargetMode, type: ActionType, targetPluginID: string, primaryArgument: any, objectType?: string): AbstractAction {
```

Comment:
Because this requires all these parameters, it's not really an abstract action, it's basically an Action and that would be backward-compatible to do so.